lib/lib.h

```diff
@@ -242,18 +242,24 @@ unsigned char *mb58Encode(const unsigned char *msg, int msg_len, int *offset) {
   while (BN_cmp(bn_dv, bn0) > 0) {
     if (!BN_copy(temp, bn_dv))
       error("unable to copy bn_dv to temp");
-
-     // printf("%lu / %lu = ", *temp->d, *bn58->d);
-    if (!BN_div(bn_dv, rem, temp, bn58, ctx))
+    // printf("%lu / %lu = ", *temp->d, *bn58->d);
+    if (!BN_div(bn_dv, bn_rem, temp, bn58, ctx))
       error("Unable to perform BIGNUM division");
 
-     // printf("bn_dv %lu rem: %lu\n", *bn_dv->d, *rem->d);
+    // printf("bn_dv %lu bn_rem: %lu\n", *bn_dv->d, *bn_rem->d);
 
-    BN_bn2bin(rem, &bin_rem); 
```

Comment:
The issue here was that converting a bignumber of zero into binary caused unexpected results. Solution was to manually write zero instead of converting bignumber to zero.